client/me/purchases/cancel-privacy-protection/index.jsx

```diff
@@ -74,8 +74,8 @@ class CancelPrivacyProtection extends Component {
 			return true;
 		}
 
-		const { selectedSite } = props,
-			purchase = getPurchase( props );
+		const { selectedSite } = props;
+		const purchase = props.selectedPurchase;
```

Comment:
You could destructure and alias props here (and pretty much everywhere else in this PR)

```js
const { selectedPurchase: purchase } = props;
```

which might make those recurring props definitions a bit more future-proof (in case someone else wants to add more props).

Not sure it's worth the hassle tho.

Reply:
Yep, this was blocked by #24862, I implemented that here but really bloated this PR. We'll follow up with #24895 which will clean up the names and make them more consistent.